recorder/src/main/cpp/profile_writer.cc

```diff
@@ -55,3 +56,129 @@ void ProfileWriter::append_wse(const recording::Wse& e) {
+            ti->set_is_daemon(info->is_daemon);
+            ti->set_tid(info->tid);
+            ss->set_thread_id(thd_id);
+        } else {
+            ss->set_thread_id(known_thd->second);
+        }
+    } else {
+        ss->set_error(translate_forte_error(trace.num_frames));
+    }
+
+    SPDLOG_DEBUG(logger, "Ctx-len count: {}", ctx_len);
+
+    for (auto i = 0; i < ctx_len; i++) {
+        auto trace_pt = ctx->at(i);
+        auto known_ctx = known_ctxs.find(trace_pt);
+        if (known_ctx == known_ctxs.end()) {
+            CtxId ctx_id = next_ctx_id++;
+            known_ctxs.insert({trace_pt, ctx_id});
+            auto new_ctx = idx_dat->add_trace_ctx();
+            new_ctx->set_trace_id(ctx_id);
+            std::string name;
+            bool is_generated;
+            std::uint8_t coverage_pct;
+            PerfCtx::MergeSemantic m_sem;
+            reg.resolve(trace_pt, name, is_generated, coverage_pct, m_sem);
+            new_ctx->set_is_generated(is_generated);
+            if (! is_generated) {
+                new_ctx->set_coverage_pct(coverage_pct);
+                new_ctx->set_merge(static_cast<recording::TraceContext_MergeSemantics>(static_cast<int>(m_sem)));
+            }
+            new_ctx->set_trace_name(name);
+            ss->add_trace_id(ctx_id);
+            SPDLOG_DEBUG(logger, "Reporting trace named '{}', cov {}% as ctx-id: {}", name, coverage_pct, ctx_id);
+        } else {
+            ss->add_trace_id(known_ctx->second);
+        }
+    }
+
+    ss->set_snipped(trace.num_frames > trunc_thresholds.cpu_samples_max_stack_sz);
+
+    if (trace.num_frames < 0) return;
+
+    for (auto i = 0; i < Util::min(static_cast<TruncationCap>(trace.num_frames), trunc_thresholds.cpu_samples_max_stack_sz); i++) {
+        auto f = ss->add_frame();
+        auto& jvmpi_cf = trace.frames[i];
+        //find method
+        auto mth_id = jvmpi_cf.method_id;
+        if (known_methods.count(reinterpret_cast<MthId>(mth_id)) == 0) {
+            if (! fir(mth_id, jvmti, *this)) {
+                recordNewMethod(mth_id, "?", "?", "?", "?");
+            }
+        }
+        //end find method
+        f->set_method_id(reinterpret_cast<std::int64_t>(mth_id));
+        f->set_bci(jvmpi_cf.lineno);//turns out its actually BCI
+        f->set_line_no(lnr(jvmpi_cf.lineno, mth_id, jvmti));
+    }
+}
+
+void ProfileSerializingWriter::recordNewMethod(const jmethodID method_id, const char *file_name, const char *class_name, const char *method_name, const char *method_signature) {
+    known_methods.insert(reinterpret_cast<MthId>(method_id));
+    auto idx_dat = cpu_sample_accumulator.mutable_indexed_data();
+    auto mi = idx_dat->add_method_info();
+    mi->set_method_id(reinterpret_cast<std::int64_t>(method_id));
+
+    assert(file_name != nullptr);
+    mi->set_file_name(file_name);
+
+    assert(class_name != nullptr);
+    mi->set_class_fqdn(class_name);
+
+    assert(method_name != nullptr);
```

Comment:
No, that will lead to a JVM crash in a assert-enabled build. We just should mark things as "unknown" perhaps.

Because JVMTI doc doesn't talk about these cases in a lot of detail.